### DISCUSSION TITLE
Moving to hourly usage-based billing for databases, based on disk consumption

### DISCUSSION THREAD

**Discussion by kevcodez** [supabase/supabase#28849] [Changelog]
**tldr:**

- **No changes for Free Plan users**
- **Billing for paid plan organizations will be based on provisioned disk rather than used database space:**
    - Each project starts with 8 GB disk provisioned by default.
    - The first 8 GB of provisioned disk per project is free, then $0.125 per additional GB.
    - Charges are prorated down to the hour, which is advantageous for short-lived projects and branches.
    - Provisioned disk from Read Replicas will also be included in billing.
    - Enables upcoming features for enhanced control over disk and Postgres parameters.

**Timeline**

This change will be rolled out to new customers on **August 26th, 2024** and will be gradually rolled out to existing customers shortly after.

**Changes**

We are adjusting our pricing to offer more flexibility and self-serve for developers wanting to tune their disk and Postgres configuration. For example:

- Some developers want disks with higher throughput
- Some developers want to store more than 1GB of WAL (for tools like Airbyte/PeerDB, or adding more read replicas)

To make this available we will start billing for provisioned disk size (rather than database space used). Previously, costs associated with WAL files were not directly billed but also users could not control change `max_wal_size` (default is 1GB). 

There is no action needed on your end. You will automatically be transitioned to the new billing model throughout the next couple of weeks. In case there is any change in your monthly bill, we will reach out to you proactively with additional information and give you a grace period to decrease your usage.

For customers on the Free Plan, there will be no changes; the total database space remains capped at 500MB. These adjustments only apply to customers on paid plans. The database disk will continue to autoscale when nearing capacity for paid plan customers.

|  | **Before**  | **After (August 26th, 2024)** |
| --- | --- | --- |
| **Price** | $0.125 / GB | $0.000171 / GB-Hr |
| Change | We take the average database space used for all projects, independent of how many days/hours you store the files and sum it up. | We will you based on the provisioned disk usage every hour. First 8GB per project are free. Read replicas will also incur disk costs. |
| Invoice Item | Your invoices display 'Total Database size'. | Your invoices will display 'Disk Size GB-Hrs'. |

### Example 1: Pro plan org, active for whole month

In this scenario, an Organization is on the Pro Plan with 3 active projects.

**Usage**

| Project | # Days Active  | Average Database Space Used | Provisioned Disk | After: Provisioned Disk Size GB-Hrs |
| --- | --- | --- | --- | --- |
| Project A | 30 | 25 GB | 40.5 GB | 29,160 (720 hours * 40.5 GB) |
| Project B | 30 | 10 GB | 27 GB | 19,440 (720 hours * 27 GB) |
| Project C | 30 | 5 GB | 8 GB | 5,760 (720 hours * 8 GB) |
|  |  |  |  |  |
| **Total** |  | **40 GB** |  | **54,360 GB-Hrs** |

**Billing**

|  | **Before** | **After** |
| --- | --- | --- |
| Total Usage | 40 GB  | 54,360 GB-Hrs |
| Usage Discount (Pro Plan) | (8 GB) | (17,280 GB-Hrs - first 8 GB per project included) |
| ***Billable Usage*** | **32 GB** | **37,080 GB-Hrs** |
|  |  |  |
| Price | $0.125 / GB | $0.000171 / GB-Hr |
| ***Total Cost*** | ***$4.00*** | ***$6.43*** |

### Example 2: Pro plan org, active for part of the month

In this scenario, an Organization is on the Pro Plan with 3 active projects.

**Usage**

| Project | # Days Active  | Average Database Space Used | Provisioned Disk | After: Provisioned Disk Size GB-Hrs |
| --- | --- | --- | --- | --- |
| Project A | 30 | 9 GB | 12 GB | 8,640 (720 hours * 12 GB) |
| Project B | 15 | 9 GB | 12 GB | 4,320 (360 hours * 12 GB) |
| Project C | 2 | 9 GB | 12 GB | 576 (48 hours * 12 GB) |
|  |  |  |  |  |
| **Total** |  | 27 GB |  | 13,536 GB-Hrs |

**Billing**

|  | **Before** | **After** |
| --- | --- | --- |
| Total Usage | 27 GB  | 13,536 GB-Hrs |
| Usage Discount (Pro Plan) | (8 GB) | (9,024 - first 8 GB per project included) |
| ***Billable Usage*** | **19 GB** | **4,512 GB-Hrs** |
|  |  |  |
| Price | $0.125 / GB | $0.000171 / GB-Hr |
| ***Total Cost*** | ***$2.38*** | **$0.77** |

### Where do I see my disk size?

You can see your project’s disk size in your [database settings](https://supabase.com/dashboard/project/_/settings/database) (Project Settings > Database).

<img width="1011" alt="Screenshot 2024-07-25 at 09 36 09" src="https://github.com/user-attachments/assets/f0429333-d65f-4172-bc6c-53569d1ea0a3">

### How can I resize my disk down?

Your disk size is based on your database space usage. As a first step, you need to identify current database space usage and reduce it. To see your current database space usage, head over to the built-in [“Database” project report](https://supabase.com/dashboard/project/_/reports/database). Once you have reduced your database space and want to reduce your provisioned disk, you can upgrade your Postgres version through your [project settings](https://supabase.com/dashboard/project/_/settings/infrastructure) to automatically rightsize your disk. For further information around disk management and reducing database space, please refer to our [docs](https://supabase.com/docs/guides/platform/database-size#disk-management).

### Is this going to affect my monthly bill?

If your current disk size is >8GB, this is likely going to impact you. Note that this will be gradually rolled out and you will be notified about the concrete impact on your organization and given a 3-month grace period, which gives you time to right-size your disk and minimize the impact of this change.
